shared/hooks/useDebounce.ts

```diff
@@ -0,0 +1,17 @@
+import { useState, useEffect } from 'react'
+
+const useDebounce = <T>(value: T, delay: number) => {
```

Comment:
:wrench: I think it would be nice to set a default here. As long as it is only used here, there is no need to keep it in `constants`.
```suggestion
const DEBOUNCE_TIMEOUT = 250

const useDebounce = <T>(value: T, delay = DEBOUNCE_TIMEOUT) => {
```